proto/substrait/algebra.proto

```diff
@@ -889,9 +889,17 @@ message Expression {
       // parameterizable.
       repeated Type.Parameter type_parameters = 3;
 
-      // the value of the literal, serialized using some type-specific
-      // protobuf message
-      google.protobuf.Any value = 2;
+      // the value of the literal, serialized using some type-specific protobuf message
+      // DEPRECATED: use the val oneof instead
+      google.protobuf.Any value = 2 [deprecated=true];
+
+      // a user-defined literal can be encoded in one of two ways
+      oneof val {
```

Comment:
I wanted to use `value` as the name, but that would break JSON compatibility as `value` is already used, so I've opted for `val`.

I'm open to suggestions for better names.

Reply:
You can wrap the existing value field with the oneof and still be backwards compatible (oneof enforces that only one of the optional types are set at construction time).  That would avoid the need to deprecate a field.

Reply:
Is this different from what I had in this commit https://github.com/substrait-io/substrait/pull/604/commits/21baca75c61b7f7658cdb85c6c42aa2993c1ae6e#r1496703047

Reply:
The difference would be the rename.  If you left it named value it wouldn't break anything.  By renaming it opaque you only break code since the old code would refer to it with the old name and the new code would refer to it with the new name.

Ah, but it WOULD break folks using JSON and text protobuf formats (which aren't usually legal as a communication method because of this particular requirement).  Unless we expect producers to populate both fields for a while there's not much gain in the deprecation approach.

Reply:
My preference would be the oneof but keep the value field named value instead of opaque.